File: 0937-online-stock-span/0937-online-stock-span.java
class StockSpanner {
    private Deque<Integer> priceStack;
    private Deque<Integer> spanStack;

    public StockSpanner() {
        priceStack = new ArrayDeque<>();
        spanStack = new ArrayDeque<>();
    }
    
    public int next(int price) {
        int span = 1;

        while (!priceStack.isEmpty() && priceStack.peek() <= price) {
            priceStack.pop();
            span += spanStack.pop();
        }

        priceStack.push(price);
        spanStack.push(span);

        return span;
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.next(price);
 */

File: 0937-online-stock-span/0937-online-stock-span.kt
class StockSpanner() {
    private val priceStack = ArrayDeque<Int>()
    private val spanStack = ArrayDeque<Int>()

    fun next(price: Int): Int {
        var span = 1

        while (priceStack.isNotEmpty() && priceStack.last() <= price) {
            priceStack.removeLast()
            span += spanStack.removeLast()
        }
        
        priceStack.addLast(price)
        spanStack.addLast(span)
        
        return span
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * var obj = StockSpanner()
 * var param_1 = obj.next(price)
 */